src/element-style-observer.js

```diff
@@ -266,7 +266,10 @@ export default class ElementStyleObserver {
 			transition = getComputedStyle(this.target).transition;
 		}
 
-		if (transition === "all" || transition === "none") {
+		// Don't disable transitions on properties we are observing. See https://github.com/LeaVerou/style-observer/issues/107
+		transition = transition.replace("none", "");
```

Comment:
Potential problem with this: what if a property we're observing is called `--something-none`?

Reply:
Nice catch! In that case, I think we can rely on the fact that the transition property always comes first in the computed value. So, if `transition` starts with `none ` (mind the space), we can safely remove it. How about this approach?

Reply:
Hmm but seems like it can also be exactly `none` sometimes (without a space). Maybe using a regular expression like `/^\s*none(?=\s|$)/` would be safer?

![CleanShot 2025-05-21 at 17 03 39@2x](https://github.com/user-attachments/assets/13d45937-fa97-4d07-b8b6-e02439e54767)

Reply:
I think that what @DmitrySharabin went with works just as well, even though it's less "defensive"; it looks like browsers automatically strip leading and trailing whitespace when retrieving property values with `getComputedStyle()` / `getPropertyValue()` ([it wasn't always like this](https://stackoverflow.com/questions/62750308/getpropertyvalue-includes-whitespace-of-css-formating-when-retrieving-custom-c), however).

Reply:
I believe the regex above will fail if we have `none` followed by a comma. Of course we could add a comma, but `\bnone\b` seems to cover all cases and is more performant.

Reply:
Hmm, @DmitrySharabin [went with `/^none\b/`](https://github.com/LeaVerou/style-observer/commit/4f5b0765eb610746ccfbb1e41d6e8d2b82213f45); the only difference with what you're proposing is that it assumes that there can never be anything (not even whitespace) before the `none` (which seems to be the case).

Reply:
(BTW, YIL [yesterday I learned 😄) about `\b`, super useful, thanks!)

Reply:
Oh but if `none` is followed by a `,` then just removing `none` is probably wrong, we probably want to remove the comma as well. 🤔

Reply:
Hold on (sorry for the multiple comments in a row), I don't think `none,` is possible. Even though `transition-property: none, some-other-property` is valid, it doesn't look like the shorthand `transition` supports `none` as the first transition's property, maybe because `none` has specific meaning in `transition` (which I guess is the root cause of #107?). See [this Codepen I just made](https://codepen.io/benface/pen/jEEgjpg). tl;dr: I'm pretty sure the fix in this PR is safe and I confirm that it fixes my issue (I tried it in a patch).

Reply:
> `/\bnone\b/` should do it, no?

This is what I began with. However, if we have something like `transition: --something-none`, `none` will also be removed, which we don't want.

Reply:
> Hmm, @DmitrySharabin [went with `/^none\b/`](https://github.com/LeaVerou/style-observer/commit/4f5b0765eb610746ccfbb1e41d6e8d2b82213f45); the only difference with what you're proposing is that it assumes that there can never be anything (not even whitespace) before the `none` (which seems to be the case).

Ah fair enough, I hadn't seen it, I was going by the regex in the screenshot above. That seems much better if browsers serialize transition that way.